### PR TITLE
Update Terraform data security object names

### PR DESCRIPTION
**Pull request by sajjadlateef**
Update terminology for Data Security objects to be consistent with the HashiCorp compliance docs

### What
Update architectural-details/data-security.mdx 
### Why
Update "Ingress VCS data" to "Terraform Configuration" 

Added
Sentinel Policies
Terraform Provider Credentials
Terraform Audit Trails
Cloud Data backups
Cloud copy of backups for DR 

### Screenshots

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [ ] Description links to related pull requests or issues, if any.

#### Content
- [ ] Redirects have been added to `website/redirects.js` for moved, renamed, or deleted pages.
- [ ] API documentation and the API Changelog have been updated. 
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [ ] Pages with related content are updated and link to this content when appropriate.
- [ ] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [ ] New pages have metadata (page name and description) at the top.
- [ ] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [ ] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [ ] UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [x] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/architectural-details/data-security.mdx [hunk 1]
```diff
@@ -14,12 +14,28 @@ seriously. This table lists which parts of the Terraform Cloud and Terraform Ent
 
 | Object                               | Storage      | Encrypted                |
 | :----------------------------------- | :----------- | :----------------------- |
-| Ingressed VCS Data                   | Blob Storage | Vault Transit Encryption |
-| Terraform Plan Result                | Blob Storage | Vault Transit Encryption |
-| Terraform State                      | Blob Storage | Vault Transit Encryption |
-| Terraform Logs                       | Blob Storage | Vault Transit Encryption |
+| Terraform Configuration (VCS data)   | Blob Storage | Vault Transit Encryption |
+| Private Registry modules             | Blob Storage | Vault Transit Encryption |
```
Comment (BrianMMcClain) [inline website/docs/cloud-docs/architectural-details/data-security.mdx:18]:
```suggestion
| Private Registry Modules             | Blob Storage | Vault Transit Encryption |
```

I think we actually need to update all of the tables to match our general style guide, but just updating this line to match the style used in the rest of doc.

**Review by BrianMMcClain — APPROVED** [review]

**Review by james-warren0 — APPROVED** [review]

**Review by beekus — APPROVED** [review]

**Review by leizerbeam — APPROVED** [review]

File: website/docs/cloud-docs/architectural-details/data-security.mdx [hunk 2]
```diff
@@ -14,19 +14,35 @@ seriously. This table lists which parts of the Terraform Cloud and Terraform Ent
...
 | SSH Keys                             | PostgreSQL   | Vault Transit Encryption |
 | User/Team/Organization Tokens        | PostgreSQL   | HMAC SHA512              |
 | OAuth Client ID + Secret             | PostgreSQL   | Vault Transit Encryption |
 | OAuth User Tokens                    | PostgreSQL   | Vault Transit Encryption |
 
+### Terraform Cloud Specific 
+
+| Object                               | Storage      | Encrypted                |
+| :----------------------------------- | :----------- | :----------------------- |
+| Cloud Data Backups                   | Blob Storage | SSE-S3                   |
```
Comment (eastebry) [inline website/docs/cloud-docs/architectural-details/data-security.mdx:41]:
```suggestion
| Cloud Data Backups                   | Amazon S3 | SSE-S3                   |
```

File: website/docs/cloud-docs/architectural-details/data-security.mdx [hunk 3]
```diff
@@ -14,19 +14,35 @@ seriously. This table lists which parts of the Terraform Cloud and Terraform Ent
...
+| Cloud Copy of Backups for DR         | Blob Storage | SSE-S3                   |
```
Comment (eastebry) [inline website/docs/cloud-docs/architectural-details/data-security.mdx:42]:
```suggestion
| Cloud Copy of Backups for DR         | Amazon S3 | SSE-S3                   |
```

**Review by eastebry — APPROVED** [review]
I'd recommend moving these two to "Amazon S3", rather than blob storage, to disambiguate between things that flow through archivist and things that are stored directly in s3. This will hopefully avoid confusion when readers review the lower section "Blob storage", which describes the process for storing things via archivist, but does not apply to backups.

Although s3 is ultimately where all of this data is stored, archivist performs its own application level encryption with Vault transit.

**Comment by sajjadlateef** [comment]
Update object names and add Cloud specific data

**Pull request merged**